src/models/transfers.js

```diff
@@ -434,7 +434,10 @@ function * setTransfer (transfer, requestingUser) {
   }
 }
 
-function * getFulfillment (transferId) {
+function * getFulfillment (transferId, requestingUser) {
+  if (!requestingUser || !requestingUser.name) {
+    throw new UnauthorizedError('Not authorized')
```

Comment:
I decided not to for now because it would require an extra database query and this is just to avoid DOS. It might be something we want in the future, I'm not sure.